### DISCUSSION TITLE
Can we make background-threaded crypto optional?

### DISCUSSION THREAD

**Discussion by mceachen** [panva/paseto#18] [Ideas]
Having `crypto_worker.js` use `worker_threads` can be slow on Windows, and doesn't work without tweaks when packaged in an asar bundle.

Would you accept a PR to make threaded crypto operations be optional?

**Comment by panva** [comment]
Actually, i don't think it's needed. I'll be eventually getting rid of worker_thread altogether as there are alternatives in current (or coming) node.js now that push these operations to a libuv threadpool

- v1.local - encryption and hmac (fast, probably spends same time serializing worker data as it does crypto), can use node's webcrypto API (nonblocking) under the covers when available
- v1.public - RSASSA-PSS - [soon async](https://github.com/nodejs/node/commit/25985d6522423fc1be9a12b11379c07b846dac90)
- v2.public - Ed25519 - [soon async](https://github.com/nodejs/node/commit/25985d6522423fc1be9a12b11379c07b846dac90)

**Reply by mceachen** [reply]
That's great, but electron trails node improvements by 6 months to a year. I'd like to ship this afternoon, if possible.

I'll just fork a bodge-job that strips out the worker thread.

Thanks again for your work here!

**Reply by panva** [reply]
I'm not gonna get to said refactor that fast. And i wouldn't accept a PR i know is going to be obsolete eventually either.

**Reply by mceachen** [reply]
> I'm not gonna get to said refactor that fast

Of course! Sorry if you thought I was implying you needed to do anything on a timetable!

I'll make do with my [fork](https://github.com/mceachen/paseto/commit/a90a6af66b50bcd6897bb1dd6416240d9c9f28c2).

**Comment by panva** [comment]
#19 that is the major v3.x bump to be released shortly gets rid of the dependency on worker_threads.
